src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.Api/Controllers/DataController.cs

```diff
@@ -272,6 +304,39 @@ public class DataController : ControllerBase
             }
         }
 
+        /// <summary>
+        /// Runs calculations on the provided data object of the defined defined data type
+        /// </summary>
+        /// <param name="dataType">The data type id</param>
+        /// <returns>Return the updated data object</returns>
+        [Authorize]
+        [HttpPut]
+        [DisableFormValueModelBinding]
+        [RequestSizeLimit(REQUEST_SIZE_LIMIT)]
+        [ProducesResponseType(typeof(DataElement), 200)]
+        [Route("{org}/{app}/v1/data")]
+        public async Task<ActionResult> Put([FromQuery] string dataType)
+        {
+            string classRef = _appResourcesService.GetClassRefForLogicDataType(dataType);
+
+            ModelDeserializer deserializer = new ModelDeserializer(_logger, _altinnApp.GetAppModelType(classRef));
+            object appModel = await deserializer.DeserializeAsync(Request.Body, Request.ContentType);
+
+            if (!string.IsNullOrEmpty(deserializer.Error))
+            {
+                return BadRequest(deserializer.Error);
+            }
+
+            if (appModel == null)
+            {
+                return BadRequest("No data found in content");
```

Comment:
Is BadRequest the correct response here or should it be a 404 Not found?

Reply:
I would think 400 is the correct response as the problem is with the content they have provided, rather than us not finding the resource that they are requesting

Reply:
I'm not entirely sure given that they provide a valid url and a correct specified dataType (which can be any string basically) and we can't find it for some reason (either because it has a typo or just doesn't exist). But the 404 response is perhaps more a bit earlier in the code if `_altinnApp.GetAppModelType(classRef)` doesn't return anything. If the classRef is found and the deserializer fails for some reason then it's on us and not the user since that would probably be malformed data. 

I guess I'm a bit to hung on stuff like this and it's dangerous to start googling stuff like this because you end up en endless debates back and forth, found this though: https://www.codetinkerer.com/2015/12/04/choosing-an-http-status-code.html. It's a clean and nice summary showing when to use which status code, then I got to the end: 

> I’m not completely sure they do matter.
> 
> There’s a lot of smart people at Facebook and they built an API that only ever returns 200.

That's when I decided I should probably stop googling and leave it up to you:)